superbench/benchmarks/base.py

```diff
@@ -71,8 +71,12 @@ def parse_args(self):
         try:
             args, unknown = self._parser.parse_known_args(self._argv)
         except BaseException as e:
-            logger.error('Invalid argument - benchmark: {}, message: {}.'.format(self._name, str(e)))
-            return False, None, None
+            if ignore_required and isinstance(e, SystemExit):
```

Comment:
Looks like we cannot tell the difference among 3 cases: `wrong argument specified`, `required argument not given` and the combination of both.
Maybe a better way could be letting benchmark specify required parameters in an additional `required_parameters` dict, as a parameter of `register_benchmark`. `register_benchmark` can then use that dict with existing `parameters` to pass `parse_args`'s check, but still using `parameters` to fill predefined_args.
What do you think?

Reply:
We can checking the wrong cases when run benchmarks, not during the registration. Just like lazy checking.

Reply:
When run the benchmark, sb will call the `parse_args` with `ignore_required=False`, which can cover all the wrong cases.

Reply:
Actually I mean whether `SystemExit` exception is the difference between `wrong argument specified` and `required argument not given`? If yes I think it's ok.

Reply:
There has one option `exit_on_error` for `argparse` module. If False, only `required argument not given` raise the `SystemExit` exception. But the default value is True, means all wrong case will raise the `SystemExit` exception:
https://github.com/python/cpython/blob/3.10/Lib/argparse.py#L2091
https://github.com/python/cpython/blob/3.10/Lib/argparse.py#L1852

What I mean is that, the argument checking is not must when do benchmark registration, if the checking can be done earlier, we can just do it. Otherwise, the checking will be done again when launch the benchmark. Superbench can handle all the wrong cases and give the error logs.

Reply:
Get your point about delaying the parameter checking for `required param` case. But what if a benchmark has `predefine_param` and `required param` at the same time?

Reply:
Looks `exit_on_error` is introduced with python>=3.9. I have updated the log description to combine the `wrong argument specified` and `required argument not given` error, we don't need to distinguish them when `ignore_invalid=True`, could you please check again?

Reply:
Just checked that, please check my latest comment. Thanks!

Reply:
In current implementation, user can set `required params` in `predefine params`  or set them in config files. If errors happen when register the benchmark, we can ignore the errors and check combined params (predefine params + params in config file) again when launch the benchmark.
For the params checking/parsing when registration, it is mainly for the function `get_all_benchmark_predefine_settings`, which is an interface to get all the predefine/default params of benchmarks.